pkg/ipam/state.go

```diff
@@ -0,0 +1,417 @@
+// Copyright 2016-2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipam
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/k8s"
+)
+
+// This submodule extends the cilium IPAM by providing a persistent storage
+// for storing IP states.
+// The main purpose is to support fixed IP pods for statefulsets.
+
+// EnableFixedIP flag indicates whether fixed IP is enabled
+var EnableFixedIP bool = true
+
+const (
+	enableFixedIPEnv string = "ENABLE_FIXED_IP"
+	ipamStateDir     string = "/var/lib/ipamstate"
+	minFileNameLen   int    = 9 // format: <ns>-<pod>:<ip>
```

Comment:
please do `minFileNameLen int = len("<ns>-<pod>:<ip>")` or at least an example that contains the minimal filename length

Reply:
Done.